src/components/ChatEntry.js

```diff
@@ -1,22 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+  const source = props.sender === 'Vladimir' ? 'local' : 'remote';
+  const [liked, setLiked] = useState(props.liked);
+  const toggleLikeButton = () => {
+    const heart = document.getElementById(props.id);
+    setLiked(!liked);
+    if (liked) {
+      heart.textContent = '🤍';
+      props.onUpdate(-1);
+    } else {
+      heart.textContent = '❤️';
+      props.onUpdate(1);
+    }
+    setLiked(!liked);
+  };
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={`chat-entry ${source}`}>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time">
+          <TimeStamp time={props.timeStamp}></TimeStamp>
+        </p>
+        <button className="like" id={props.id} onClick={toggleLikeButton}>
+          🤍
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number,
+  sender: PropTypes.string,
+  body: PropTypes.string,
+  timeStamp: PropTypes.string,
+  liked: PropTypes.bool,
```

Comment:
If these were `.isRequired` in `ChatLog`, they probably should be here too. Now, that being said I know that causes a few test failures, but that's due to our test data not setting these. Either way, completely understand why you may not have set it here, given that.